### ISSUE TITLE
Question on simplex bijector implementation

### ISSUE THREAD

**Issue by Red-Portal** [TuringLang/Bijectors.jl#283]
Hi, 

It appears that `torch.probability` simply uses [softmax](https://github.com/pytorch/pytorch/blob/8472c24e3b5b60150096486616d98b7bea01500b/torch/distributions/constraint_registry.py#L246C41-L246C41) for the simplex bijector. 
Is there a reason our simplex transform is much more complicated? 
I was also thinking about a GPU-friendly implementation, which the current implementation appears hard do.

**Comment by torfjelde** [comment]
Softmax isn't bijective. The one we have now is (maps from d to d-1 dimensional)

**Comment by devmotion** [comment]
See also #51.

**Comment by Red-Portal** [comment]
Betanalpha does [discuss](https://discourse.mc-stan.org/t/jacobian-of-softmax-tansformation-of-a-n-1-degrees-of-freedom-unbounded-parameter/18780/26?u=red-portal) a bijective softmax by arbitrarily setting the endpoint logits. Any experience with this?

**Comment by devmotion** [comment]
That's supported e.g. in GPLikelihoods (see maybe also the discussion in https://github.com/JuliaGaussianProcesses/GPLikelihoods.jl/issues/55).

**Comment by Red-Portal** [comment]
Good to know thanks. Though, I really wish that our simplex bijector could play nicely with GPUs out of the box. Among non-NF bijectors, it seems the simplex bijector is really going to be the big challenge going in that direction. Do we have any plans on how to pursue this?

**Comment by Red-Portal** [comment]
Actually, nevermind. I just wrote a stick-breaking bijector using array operations based on the implementations of numpyro and tensorflow. If this were to be added to `Bijectors.jl` we'll probably have to add a CUDA array specialization. Let me know how to proceed on this.

**Comment by devmotion** [comment]
On Julia >= 1.9, a CUDA specialization could be put in an extension (possibly could even just be an extension with GPUArrays).

**Comment by Red-Portal** [comment]
I do have the feeling that this will have to wait until the batch operation interface is finalized. @torfjelde  Do we have an expectation on when that would be?

**Comment by penelopeysm** [comment]
> Is that raw computation time for log-density or gradient, or is that ESS/wallclock time?

It was just the time for the transform + logjac itself. I vaguely remember doing some sampling tests but it wasn't scientific enough for me to report back. These are very useful plots! I am guessing that the implementation is in Stan? I am definitely happy to open a new issue again if you're interested in translating to Julia. It probably wouldn't be my main focus, but happy to provide support with any benchmarks, etc.
